example-site/components/TableErrorValidation/TableWithValidation.tsx

```diff
@@ -0,0 +1,373 @@
+	| typeof filesPageError;
+
+// const getPageErrors = ({name}: {name: PageErrors[number]['name']}) => {
+
+// }
+
+const sampleTableError = {
+	href: '#vehicle-build-and-layout-plans-upload-button',
+	message: 'Upload files',
+	name: 'files',
+};
+type SampleTableError = typeof sampleTableError;
+
+export const TableWithValidation = ({ tableRef }: MultiTablePageProps) => {
+	const tableSectionAlertRef = useRef<HTMLDivElement>(null);
+
+	// Input data
+	const [dateValue, setDateValue] = useState<Date>();
+	const [locationValue, setLocationValue] = useState('');
+
+	// Table data
+	const [assessmentFiles, setAssessmentFiles] = useState(
+		defaultAssessmentFiles
+	);
+	const [currentAssessmentFile, setCurrentAssessmentFile] =
+		useState<AssessmentFile>();
+
+	// Errors
+	const [pageErrors, setPageErrors] = useState<PageError[]>([]);
+	const [tableErrors, setTableErrors] = useState<SampleTableError[]>([]);
+
+	// Drawer state
+	const [isDrawerOpen, openDrawer, closeDrawer] = useTernaryState(false);
+
+	// Action handlers (includes some mocking)
+	const mockUploadFile = () => {
+		const newAssessmentFiles = assessmentFiles.map((prevAssessmentFile) =>
+			currentAssessmentFile?.documentType === prevAssessmentFile.documentType
+				? {
+						...prevAssessmentFile,
+						file: 'filename.pdf',
+						size: '88kb',
+						error: false,
+				  }
+				: prevAssessmentFile
+		);
+		setAssessmentFiles(newAssessmentFiles);
+
+		setTableErrors(
+			newAssessmentFiles.reduce<SampleTableError[]>(
+				(acc, assessmentFile) =>
+					assessmentFile.file
+						? acc
+						: [
+								...acc,
+								{
+									href: `#${assessmentFile.buttonId}`,
+									message: `Upload a file for ${assessmentFile.documentType}`,
+									name: 'files',
+								},
+						  ],
+				[]
+			)
+		);
+		closeDrawer();
+	};
+
+	const mockUploadError = () => {
+		const newAssessmentFiles = assessmentFiles.map((prevAssessmentFile) =>
+			currentAssessmentFile?.documentType === prevAssessmentFile.documentType
+				? {
+						...prevAssessmentFile,
+						file: '',
+						size: '',
+						error: true,
+				  }
+				: prevAssessmentFile
+		);
+		setAssessmentFiles(newAssessmentFiles);
+		setPageErrors((prevPageErrors) => {
+			return Array.from(new Set([...prevPageErrors, filesPageError]));
+		});
+		setTableErrors(
+			newAssessmentFiles.reduce<SampleTableError[]>(
+				(acc, assessmentFile) =>
+					assessmentFile.file
+						? acc
+						: [
+								...acc,
+								{
+									href: `#${assessmentFile.buttonId}`,
+									message: `Upload a file for ${assessmentFile.documentType}`,
+									name: 'files',
+								},
+						  ],
+				[]
+			)
+		);
+		closeDrawer();
+	};
+
+	const removeFile = () => {
+		setAssessmentFiles((prevAssessmentFiles) =>
+			prevAssessmentFiles.map((prevAssessmentFile) =>
+				currentAssessmentFile?.documentType === prevAssessmentFile.documentType
+					? {
+							...prevAssessmentFile,
+							file: '',
+							size: '',
+							error: true,
```

Comment:
I'm not sure we should be valiadating this an invalid & File missing in the table when someone removes it? Might be worth checking with the designers to see when we validate this. Should it only be on submit?

Reply:
Hmm good question, might add this to a jam session. I'm not sure either, now.

- [ ] team jam re: process of remove file